rt_entt_codegen/shared/parallel.cc

```diff
@@ -0,0 +1,233 @@
+#include "parallel.hh"
+
+#include <set>
+
+#include "ecsact/lang-support/lang-cc.hh"
+#include "ecsact/cpp_codegen_plugin_util.hh"
+
+struct loop_iterator_fields {
+	std::vector<ecsact_system_like_id>&                system_list;
+	const std::vector<ecsact_system_like_id>::iterator begin;
+	std::vector<std::vector<ecsact_system_like_id>>&   parallel_system_cluster;
+};
+
+auto loop_iterator(
+	std::vector<ecsact_system_like_id>&                system_list,
+	const std::vector<ecsact_system_like_id>::iterator begin,
+	std::vector<std::vector<ecsact_system_like_id>>&   parallel_system_cluster
+) -> void;
+
+auto ecsact::rt_entt_codegen::parallel::print_parallel_system_executions(
+	ecsact::codegen_plugin_context&                     ctx,
+	const ecsact::rt_entt_codegen::ecsact_entt_details& details,
+	std::vector<ecsact_system_like_id>                  system_list,
+	bool                                                is_child
+) -> void {
+	using ecsact::cc_lang_support::cpp_identifier;
+	using ecsact::cpp_codegen_plugin_util::block;
+
+	auto parallel_system_cluster =
+		std::vector<std::vector<ecsact_system_like_id>>{};
+
+	loop_iterator(system_list, system_list.begin(), parallel_system_cluster);
+
+	for(const auto& systems_to_parallel : parallel_system_cluster) {
+		if(systems_to_parallel.size() == 1) {
+			auto sync_sys_id = systems_to_parallel[0];
+
+			auto sync_sys_name =
+				cpp_identifier(ecsact::meta::decl_full_name(sync_sys_id));
+
+			if(details.is_action(sync_sys_id)) {
+				if(is_child) {
+					ctx.write(std::format(
+						"ecsact::entt::execute_actions<{}>(registry, parent_context, "
+						"actions_map);\n",
+						sync_sys_name
+					));
+				} else {
+					ctx.write(std::format(
+						"ecsact::entt::execute_actions<{}>(registry, nullptr, "
+						"actions_map);\n",
+						sync_sys_name
+					));
+				}
```

Comment:
This could be greatly simplified if we add the parent context variable name as a parameter instead of the `is_child` bool.
```suggestion
				ctx.write(std::format(
					"ecsact::entt::execute_actions<{}>(registry, {}, "
					"actions_map);\n",
					sync_sys_name,
					parent_context_var,
				));
```